llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -382,6 +382,51 @@ static void ComputePTXValueVTs(const TargetLowering &TLI, const DataLayout &DL,
   }
 }
 
+static EVT getVectorizedVT(EVT VT, unsigned N, LLVMContext &C) {
```

Comment:
This could use a comment. 

We return an EVT that can hold N VTs
If the VT is a vector, the resulting EVT is a flat vector with the same element type as VT's element type (as opposed to N-element vector of VT vectors).

If may be better for readability to restructure the code so we don't need to explain it as a comment:
```
   return VT.isVector()
        ? EVT::getVectorVT(C, VT.getScalarType(), VT.getVectorNumElements()  * N)
        : EVT::getVectorVT(C, VT, N);
```

Reply:
Nice. Switched to doing it like this.